enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,848 @@
+by combining the `annotation_prefix`, the key from
+`resources`, and the container name, like this:
+
+```text
+io.openshift.workload.management.cpushares/container_name = {value}
+```
+
+In the management workload case, we will configure it with values like
+
+```ini
+[crio.runtime.workloads.management]
+  activation_annotation = "workload.openshift.io/management"
+  annotation_prefix = "io.openshift.workload.management"
+  resources = { "cpushares" = "", "cpuset" = "0-1" }
+```
+
+CRI-O will be configured to support a new annotation on pods,
+`io.openshift.workload.management.cpushares/{container-name}`.
+
+```ini
+[crio.runtime.runtimes.runc]
+  runtime_path = "/usr/bin/runc"
+  runtime_type = "oci"
+```
+
+Pods that have the `workload.openshift.io/management` annotation will
+have their cpuset configured to the value from the appropriate
+workload configuration. The CPU shares for each container in the pod
+will be configured to the value of the annotation with the name
+created by combining the `annotation_prefix`, `"cpushares"`
+and the container name (for example,
+`io.openshift.workload.management.cpushares/my-container`).
+
+Note that this field does not conflict with the `infra_ctr_cpuset`
+config option, as the infra container will still be put in that
+cpuset.  They can be configured as the same value if the infra
+container should also be considered to be managed.
+
+#### API Server Admission Hook
+
+A new admission hook in the kubernetes API server within OpenShift
+will mutate pods when they are created to make 3 changes.
+
+1. It will move CPU requests to workload CPU requests, so that
+   the scheduler can successfully place the pod on the node, even
+   though the sum of the CPU requests for all management workloads may
+   exceed the actual CPU capacity of the management CPU pool.
+2. It will strip all container-specific workload annotations from new
+   pods. This both prevents a user from configuring a pod incorrectly
+   and prevents unprivileged users from running their pods in a CPU
+   pool they should not have access to.
+3. It will add annotations with for each container with the value
+   equal to the original CPU requests * 1000, so that CRI-O can use
+   the value to configure the CPU shares for the container.
+
+We will not change pods in a way that changes their [Quality of
+Service](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/)
+class. So, we would not strip CPU requests unless they also have
+memory requests, because if we mutate the pod so that it has no CPU or
+memory requests the quality-of-service class of the pod would be
+changed automatically. Any pod that is already BestEffort
+would be annotated using `0` as the value so that CRI-O will have an
+indicator to configure the CPU shares as BestEffort.
+
+#### Kubelet Changes
+
+Kubelet will be changed to look for a configuration file,
+`/etc/kubernetes/workload-pinning`, to enable the management
```

Comment:
minor nit: can we name this file with an openshift domain prefix or suffix?

`/etc/kubernetes/openshift-workload-pinning`

Reply:
> It's disappointing that we won't have installer support this go-round, but understandable. We should track this part as a GA requirement for a future release.

https://github.com/openshift/enhancements/pull/727

Reply:
#726 updates the name